### PR TITLE
chore(bot): change Docker base image to python:3.7-slim

### PR DESCRIPTION
**Pull request by etiennetremel**
While scanning the Kodiak Docker image with [Grype](https://github.com/anchore/grype), we noticed a significant amount of vulnerabilities.

This PR upgrade the base image to the latest 3.7 image and use the slim version which come with a smaller size and reduce the attack surface. I also took the opportunity to use a non root user.

- previously: image size 1.31GB, 3694 vulnerabilities
- now: image size 607Mb, 87 vulnerabilities

### CODE REVIEW

**Comment by sbdchd** [comment]
Nice, I love that smaller image size!

**Comment by etiennetremel** [comment]
Anything else I can help with?

**Comment by chdsbd** [comment]
@etiennetremel It looks the CI job for building the bot container failed. Should be good to merge once that's fixed

**Comment by etiennetremel** [comment]
ah, that's interesting.. it worked on my machine 😛
I made a few changes, let's see how that goes

**Comment by etiennetremel** [comment]
@chdsbd anychance you could run the docker build job with credentials? 
image

File: bot/Dockerfile
Comment (sbdchd) [inline bot/Dockerfile:12]: Don’t we still need the supervisor conf?

Comment (etiennetremel) [reply bot/Dockerfile:12]: Good catch, I just added that missing line and also updated supervisord.conf to redirect stderr logs to stdout. It makes it easier to debug.

File: bot/Dockerfile
Comment (sbdchd) [inline bot/Dockerfile:1]: looks good, one minor thing is we should probably have the tests run in a python:3.7-slim container instead of the current python:3.7 to head off any differences before release 

Also @chdsbd probably wants to run these containers on staging before merge although I don't think we use any OS specific stuff

Comment (etiennetremel) [reply bot/Dockerfile]: good point, I just updated the container to use slim version for the bot test step in the circleci config

**Comment by chdsbd** [comment]
Not sure why CI is failing

**Comment by etiennetremel** [comment]
> Not sure why CI is failing

Error was being triggered during the pip install of poetry, for this use case I reckon we can safely use the flag `--root-ignore-action=ignore` as we then use kodiak as user to run the app from supervisord.

**Comment by sbdchd** [comment]
Oh I think we need to update CI to install `git` for the script to pass:

https://app.circleci.com/pipelines/github/chdsbd/kodiak/2726/workflows/6fe5f10b-3fc1-4ecd-a1ea-faf4347e28c1/jobs/25425

**Comment by novascreen** [comment]
@etiennetremel @sbdchd any change you could give this another look? it would be really nice to get vulnerabilities resolved

**Comment by etiennetremel** [comment]
@novascreen it's pretty old but I just rebased the branch. The Docker credentials are missing in CircleCI, I reckon only @sbdchd or @chdsbd would be able to help with it.

File: bot/Dockerfile
Comment (chdsbd) [inline bot/Dockerfile:23]: This appears to error in CI:

```
Step 5/9 : COPY --chown=kodiak pyproject.toml poetry.lock /var/app
When using COPY with more than one source file, the destination must be a directory and end with a /

Exited with code exit status 1
```

https://app.circleci.com/pipelines/github/chdsbd/kodiak/2818/workflows/f4a433c9-18a4-4a29-9873-7e601d2e5a83/jobs/26706

Comment (etiennetremel) [reply bot/Dockerfile]: ahh, good point, I just updated it

**Review by chdsbd — APPROVED** [review]
Thank you!

**Comment by novascreen** [comment]
Awesome, thank you!

**Pull request merged**
